packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -3,13 +3,18 @@ import PropTypes from 'prop-types';
 import 'terra-base/lib/baseStyles';
 import Button from 'terra-button';
 import IconCalendar from 'terra-icon/lib/icon/IconCalendar';
+import AppDelegate from 'terra-app-delegate';
```

Comment:
Good stuff Ben! The one thing that I'd change is the inclusion of the AppDelegate at all in the DateInput component; I'd prefer to see the request/release functions passed into this separately.

The AppDelegate has a number of functions that are of no concern to the DateInput component, and I don't want to open the door to any view component having an AppDelegate prop in the future and muddying the waters of what a component's responsibilities are.  

This just means that the component consuming the DateInput (that _is_ receiving an AppDelegate prop) would have to pass in those two functions from the AppDelegate into the DateInput separately.